src/ewatercycle/models.py

```diff
@@ -11,23 +11,89 @@
 
     To list all available models:
 
-    >>> from ewatercycle.models
-    >>> dir(lambaewatercycle.models)
-    # List models + some private stuff
-
-    To list all available models in an ipython shell
-
-    >>> from ewatercycle.models import
-    # Followed by pressing the TAB key and picking a model from the autocomplete list.
+    >>> from ewatercycle.models import sources
+    >>> sources
+    # Returns dictionary containing models
 
 To get your own model to be listed here it needs to be
 registered in the :py:data:`ewatercycle.models`
 `entry point group <https://packaging.python.org/en/latest/specifications/entry-points/>`_.
 """
+
 from importlib.metadata import entry_points
+from typing import Mapping, Type
+
+from importlib_metadata import EntryPoint
+
+from ewatercycle.base.model import eWaterCycleModel
 
 _model_entrypoints = entry_points(group="ewatercycle.models")  # /NOSONAR
 
 # Expose as "from ewatercycle.models import Model" for backward compatibility
 for _model in _model_entrypoints:
     globals()[_model.name] = _model.load()
+
+
+class ModelSources(Mapping):
```

Comment:
They are very similar yes. In some way it would be nicer to be able to list models, forcings, and parameter sets all in the same way. 

Before this change it was:
- `ewatercycle.forcing.sources`
- `dir(ewatercycle.models)` _which also was cluttered with non-useful info_
- `ewatercycle.parameter_sets.available_parameter_sets()`

It would also be nice to still unify how to access these... three different ways for what is essentially the same is not very user friendly.

Reply:
> Can we have the same super? Can we make a Source super class with a generic?

Done.

> It would also be nice to still unify models, forcings, and parameter

I've now unified models and forcings, however parameter_sets it still tbd. Probably better in a separate PR.

Reply:
Would be nice if you did not need a subclass with __getitem__  method.
I tried `Mapping[str, DefaultForcing]` and [UserDict](https://docs.python.org/3/library/collections.html#userdict-objects) instead of Mapping, but got stuck on value being a class or an entrypoint.

On my machine it does not make much difference if load() is called or not:
```ipython
In [1]: from importlib.metadata import entry_points
In [4]: %timeit [f.load() for f in entry_points(group="ewatercycle.forcings")]
7.33 ms ± 158 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [5]: %timeit entry_points(group="ewatercycle.forcings")
7.3 ms ± 81.8 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```
So we could simplify by calling the load() on import.

Reply:
The subclassing here is to get the typing correct, not to delay the loading.

for the model sources:
```py
    def __getitem__(self, key) -> Type[eWaterCycleModel]:
```

while the forcing sources are:
```py
    def __getitem__(self, key) -> Type[DefaultForcing]:
```